Python/ceval.c

```diff
@@ -44,6 +44,23 @@
 #  error "ceval.c must be build with Py_BUILD_CORE define for best performance"
 #endif
 
+#if !defined(Py_DEBUG)
+// bpo-45116: The MSVC compiler fails to inline these in PGO build,
+// and they're kind of important for performance.
+
+#undef Py_DECREF
+#define Py_DECREF(arg) do { PyObject *op = _PyObject_CAST(arg); if (--op->ob_refcnt == 0) { destructor d = Py_TYPE(op)->tp_dealloc; (*d)(op); } } while (0)
+
+#undef Py_IS_TYPE
+#define Py_IS_TYPE(ob, type) (_PyObject_CAST(ob)->ob_type == (type))
+
+#undef Py_XDECREF
+#define Py_XDECREF(arg) do { PyObject *op1 = _PyObject_CAST(arg); if (op1 != NULL) { Py_DECREF(op1); } } while (0)
+
+#endif
+
+#define _Py_atomic_load_32bit_impl(value, order) (assert((order) == _Py_memory_order_relaxed), *(value))
```

Comment:
I notice the original signature is this:

```c
inline int _Py_atomic_load_32bit_impl(volatile int* value, int order)
```

Don't we need some sort of cast to `volatile int` so there's no risk of strange compiler optimizations?

Reply:
Never mind, it looks okay: things are declared as `_Py_atomic_int` which uses `volatile int _value;`, so everything should be okay.

Reply:
I do wonder why we are going through the hassle of these macros. Either @vstinner or @markshannon, do you remember what the concern here was? Why not just declare these are `volatile int32` (or `_Py_atomic_int`) and be done? All the callsites in this file specify "relaxed" mode. Are there platforms where things must be done differently?

Reply:
Some answers here:
https://stackoverflow.com/questions/6627596/is-volatile-int-in-c-as-good-as-stdatomicint-of-c0x

Apparently, volatile variables are not required to be atomic.

Reply:
I'm sorry, I wasn't arguing for `volatile` instead of `_Py_atomic_int`. But since, with the `_Py_atomic_int` declaration and the `_Py_memory_order_relaxed` memory order, apparently the whole `_Py_atomic_load_relaxed(x)` call degenerates to `*x`, why have a macro or inline function at all?

Reply:
The more I look at pycore_atomic.h the more I fail to believe that anybody knows what it does. And even more so for dynamic_annotations.h. :-(

Anyway, I should probably override just `_Py_atomic_load_relaxed`, not `_Py_atomic_load_32bit_impl`. The former is more specific. And I should move it back inside the `#if !defined(Py_DEBUG)` block. (Hm, is `#ifndef` out of style?)